### PR TITLE
snap-repair: implement `snap-repair {done,skip,retry}`

### PR DESCRIPTION
**Pull request by mvo5**
The snap-repairs will use `snap-repair {done,skip,retry}` to
signal the status to the repair runner. The command in turn will
write the right status to the SNAP_REPAIR_STATUS_FD but that is
an implementation detail that should not leak into the repair
scripts itself.

### CODE REVIEW

File: cmd/snap-repair/cmd_done_retry_skip.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+func init() {
+	cmd, err := parser.AddCommand("done", "Signal repair is done", "", &cmdDone{})
+	if err != nil {
+
+		panic(err)
+	}
+	cmd.Hidden = true
+
+	cmd, err = parser.AddCommand("skip", "Signal repair is done", "", &cmdSkip{})
```
Comment (pedronis) [inline cmd/snap-repair/cmd_done_retry_skip.go]: s/is done/should be skipped/ ?

File: cmd/snap-repair/cmd_done_retry_skip.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	if err != nil {
+		panic(err)
+	}
+	cmd.Hidden = true
+
+	cmd, err = parser.AddCommand("retry", "Signal repair is done", "", &cmdRetry{})
```
Comment (pedronis) [inline cmd/snap-repair/cmd_done_retry_skip.go]: s/is done/must be retried next time/

File: cmd/snap-repair/cmd_done_retry_skip.go (hunk 1 above)
Comment (mvo5) [reply cmd/snap-repair/cmd_done_retry_skip.go]: :man_facepalming: Thanks!

**Comment by pedronis** [comment]
Also I now noticed that in the topic we decided that the command to run inside the scripts would just be 'repair' by using some symlink.

**Comment by mvo5** [comment]
@pedronis https://github.com/snapcore/snapd/pull/3933 implements one possible way to make the `repair` binary available for scripts. The other one would be a permanent symlink in /usr/lib/snapd/ but then we also make some internal snapd commands available to the script which feels not ideal.

**Pull request merged**